[WIP] feat: JAM-NP CE Handler

nu1lspaxe opened the pull request:
Only focus on `internal/networking/handler/ce`, other files are auto included.

mtsay committed:
Applied graypaper 0.6.6 updates.

mtsay committed:
Applied graypaper 0.6.6 updates.

mtsay committed:
Fix statistics tests and others.

Also removed the ability to update AuthPoolMaxSize and
AuthQueueSize.

yu2C committed:
refactor: update account type

refactor: update account type

yu2C committed:
refactor: update account type

refactor: update account type

yu2C committed:
refactor: update account type

refactor: update account type

yu2C committed:
refactor: update account type

refactor: update account type

yu2C committed:
refactor: update account type

refactor: update account type

yu2C committed:
refactor: update account type

refactor: update account type

mtsay committed:
Clean up TODOs.

mtsay committed:
Clean up TODOs.

mtsay committed:
Clean up TODOs.

mtsay committed:
Clean up TODOs.

mtsay committed:
Clean up TODOs.

mtsay committed:
Clean up TODOs.

HanaYukii committed:
feat: Update accumulation GP-067

HanaYukii committed:
feat: Update accumulation GP-067

HanaYukii committed:
feat: Update accumulation GP-067

HanaYukii committed:
feat: Update accumulation GP-067

HanaYukii committed:
feat: Update accumulation GP-067

HanaYukii committed:
feat: Update accumulation GP-067

HanaYukii committed:
Add todo

HanaYukii committed:
Add todo

HanaYukii committed:
Add todo

HanaYukii committed:
Add todo

HanaYukii committed:
Add todo

HanaYukii committed:
Add todo

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
feat: update WP to GP 0.6.7

weigen393 committed:
fix: gas type change

weigen393 committed:
fix: gas type change

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

mtsay committed:
Made corrections per review feedback.

nu1lspaxe committed:
fix: submodules update access failed due to http way

nu1lspaxe committed:
fix: submodules update access failed due to http way

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
resolve conflicts

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

TwEricShen committed:
refactor: update PVM(independent) GP 0.6.7

YCC3741 committed:
doc: Update semantic-commit-messages.md

YCC3741 committed:
doc: Update semantic-commit-messages.md

YuChunTsao committed:
fix: Update state_key_constructor (D.1)

YuChunTsao committed:
fix: Update state_key_constructor (D.1)

YuChunTsao committed:
fix: Update state_key_constructor (D.1)

YuChunTsao committed:
fix: Update state_key_constructor (D.1)

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

TwEricShen committed:
refactor: update PVM invocations GP 0.6.7

nu1lspaxe committed:
Revert "fix: submodules update access failed due to http way"

This reverts commit 314ef3db4f58528293b647f8b2091aa7890f9d5d.

nu1lspaxe committed:
Revert "fix: submodules update access failed due to http way"

This reverts commit 314ef3db4f58528293b647f8b2091aa7890f9d5d.

nu1lspaxe committed:
feat: alternative name function used in jam-np, tests

nu1lspaxe committed:
feat: alternative name function used in jam-np, tests

nu1lspaxe committed:
docs: remove unused comment

nu1lspaxe committed:
docs: remove unused comment

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

yu2C committed:
refactor: Recent history and LastAccOut type and funcs, remain test

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
fix bugs

TwEricShen committed:
feat: JIP-1 log host-call

TwEricShen committed:
feat: JIP-1 log host-call

TwEricShen committed:
chore: remove host call exception from HostCallfunctions

TwEricShen committed:
chore: remove host call exception from HostCallfunctions

TwEricShen committed:
feat: host call logger

TwEricShen committed:
feat: host call logger

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
refactor: mmr_superPeak func should output OpaqueHash

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: revert beefyBelt type

yu2C committed:
chore: OpaqueHash unmarshal is string

yu2C committed:
chore: OpaqueHash unmarshal is string

yu2C committed:
chore: OpaqueHash unmarshal is string

yu2C committed:
chore: OpaqueHash unmarshal is string

yu2C committed:
chore: OpaqueHash unmarshal is string

yu2C committed:
chore: OpaqueHash unmarshal is string

YuChunTsao committed:
Merge pull request #580 from New-JAMneration/559-update-recent-history-gp-067

refactor: Recent history and LastAccOut type and funcs, remain test

YuChunTsao committed:
Merge pull request #580 from New-JAMneration/559-update-recent-history-gp-067

refactor: Recent history and LastAccOut type and funcs, remain test

YCC3741 committed:
fix: merge conflicts dealed

YCC3741 committed:
fix: merge conflicts dealed

YuChunTsao committed:
fix: Update state serialization

YuChunTsao committed:
fix: Update state serialization

YuChunTsao committed:
fix: Update state serialization

YuChunTsao committed:
fix: Update state serialization

YuChunTsao committed:
chore: remove deprecated serialization functions

YuChunTsao committed:
chore: remove deprecated serialization functions

YuChunTsao committed:
chore: remove deprecated serialization functions

YuChunTsao committed:
chore: remove deprecated serialization functions

YuChunTsao committed:
fix: Update state delta lookup serialization

YuChunTsao committed:
fix: Update state delta lookup serialization

YuChunTsao committed:
fix: Update state delta lookup serialization

YuChunTsao committed:
fix: Update state delta lookup serialization

YCC3741 committed:
chore: refmt

YCC3741 committed:
chore: refmt

YCC3741 committed:
Merge pull request #586 from New-JAMneration/557-update-to-gp-v067

557 update to gp v067

YCC3741 committed:
Merge pull request #586 from New-JAMneration/557-update-to-gp-v067

557 update to gp v067

TwEricShen committed:
fix bug

TwEricShen committed:
fix bug

TwEricShen committed:
revise coreID & serviceID may be nil

TwEricShen committed:
revise coreID & serviceID may be nil

TwEricShen committed:
Merge branch 'main' into 578-feat-jip-1

TwEricShen committed:
Merge branch 'main' into 578-feat-jip-1

TwEricShen committed:
chore: gofmt

TwEricShen committed:
chore: gofmt

TwEricShen committed:
chore: renamed coreID to coreId

TwEricShen committed:
chore: renamed coreID to coreId

weigen393 committed:
feat: keystore interface

weigen393 committed:
feat: keystore interface

weigen393 committed:
feat: ed25519 keypair function

weigen393 committed:
feat: ed25519 keypair function

weigen393 committed:
feat: add redis keystore and unit test

weigen393 committed:
feat: add redis keystore and unit test

YCC3741 committed:
Merge pull request #584 from New-JAMneration/578-feat-jip-1

578 feat jip 1

YCC3741 committed:
Merge pull request #584 from New-JAMneration/578-feat-jip-1

578 feat jip 1

nu1lspaxe committed:
fix: remove unnessary context param

nu1lspaxe committed:
fix: remove unnessary context param

nu1lspaxe committed:
feat: apply max incoming streams

nu1lspaxe committed:
feat: apply max incoming streams

nu1lspaxe committed:
Feat: event bus, peer implementation, ce request/handler, and (basic)
sync manager

1. Event Bus: all events and sub/pub
2. Peer: not yet apply blockchain instance, but includes peer role, pk,
   genesis header, and both handlers
3. CE request/handler: for now, only has CE 128 protocol, others are
   todo
4. Sync Manager: it should with peers to sync based on finalized block

nu1lspaxe committed:
Feat: event bus, peer implementation, ce request/handler, and (basic)
sync manager

1. Event Bus: all events and sub/pub
2. Peer: not yet apply blockchain instance, but includes peer role, pk,
   genesis header, and both handlers
3. CE request/handler: for now, only has CE 128 protocol, others are
   todo
4. Sync Manager: it should with peers to sync based on finalized block

Terryhung committed:
feat: integrate blockchain into sync_manager

Terryhung committed:
feat: integrate blockchain into sync_manager

nu1lspaxe committed:
feat: (fix ce128) and ce129 encode + test

nu1lspaxe committed:
feat: (fix ce128) and ce129 encode + test

nu1lspaxe committed:
feat: ce131 + test

nu1lspaxe committed:
feat: ce131 + test

nu1lspaxe committed:
feat: ce133 + test, (fix) stream with buffer instead of reader

nu1lspaxe committed:
feat: ce133 + test, (fix) stream with buffer instead of reader

nu1lspaxe committed:
feat: ce 134,135,136,137 with test cases and shard assignment

nu1lspaxe committed:
feat: ce 134,135,136,137 with test cases and shard assignment

nu1lspaxe committed:
feat: ce 138, 139, 140, 141 with redis store

nu1lspaxe committed:
feat: ce 138, 139, 140, 141 with redis store

nu1lspaxe committed:
feat: ce142, 143, 144, 145 with test

nu1lspaxe committed:
feat: ce142, 143, 144, 145 with test

nu1lspaxe committed:
fix: add boundary node to blockchain and ce129

nu1lspaxe committed:
fix: add boundary node to blockchain and ce129

yu2C reviewed (commented):
reviewed 128~136

nu1lspaxe committed:
fix: error message and work report hash signature

nu1lspaxe committed:
fix: error message and work report hash signature

nu1lspaxe committed:
feat: implement encode for ce request

nu1lspaxe committed:
feat: implement encode for ce request

yu2C reviewed (commented):
CE 133, 134, 137, 138, 139/140 -> @weigen393 help me check if erasure root and segment root operations are aligned with GP.

Finish the first review here. Still many things need to discuss and make decisions

nu1lspaxe committed:
docs: add todo and todo-validation for ce handlers

nu1lspaxe committed:
Merge remote feat/jam-np-ce-handler, resolve add/add conflicts

nu1lspaxe committed:
Merge remote feat/jam-np-ce-handler, resolve add/add conflicts

nu1lspaxe committed:
Fix: ce handler pr comment
1. Add Role identifier
2. Fix coreindex type u32 to u16
3. Fix quic.Stream type to use streaming read
4. Fix header hash type to type.HeaderHash{}

nu1lspaxe committed:
fix: ce132 uses correct function VerifyTicketsProof

nu1lspaxe committed:
fix: ce134 message concat and work-report hash

nu1lspaxe committed:
fix: ce137 uses T(s,i,H) to construct justification

nu1lspaxe committed:
fix: ce143, preimage should be hashed

nu1lspaxe committed:
refactor: adopt quic.stream helper function

nu1lspaxe committed:
fix: use correct 11.13 for message

nu1lspaxe committed:
fix: ce133 message read/write

nu1lspaxe committed:
fix: revise ce handler role and create proper alternative for another role

nu1lspaxe committed:
fix: alternative ce handler

nu1lspaxe committed:
fix: ce 139/140/141 handler for another role

nu1lspaxe committed:
fix: ce 142/145 alternative handler

nu1lspaxe committed:
feat: ce131/ce132 validation

nu1lspaxe committed:
fix: ce133 verify length of work package

HanaYukii committed:
feat: draft for merklization change for CE129

nu1lspaxe committed:
fix: ce131 ce132 should both check proxy index

nu1lspaxe committed:
Fix: ce131 validation
1. remainingSlots should start from SlotSubmissionEnd.
2. proxy validator forwarding to more than one validators, so appends
   batch size computing.

nu1lspaxe committed:
Feat: add ce134 send side handler (+ verify ed25519 signature)

HanaYukii committed:
refactor: merge boundary paths in CE129Response

HanaYukii committed:
fix: tiny cleanup in merklization.go

nu1lspaxe committed:
Fix: check the length of nextValidators before module operation to prevent zero module

nu1lspaxe committed:
Fix: safer calculation than math.Floor and print only validator.ed25519
key

nu1lspaxe committed:
feat: ce145 ed25519 signature validation

nu1lspaxe committed:
fix: update error message

HanaYukii committed:
refactor: change the type of the path and boundary nodes from OpaqueHash to ByteSequence

nu1lspaxe committed:
Feat: ce144 validation and send side handler

nu1lspaxe committed:
Feat: add justification varification (merkle tree)

HanaYukii committed:
refactor: migrate types for ce129 to ce129.go

yu2C reviewed (commented):
Small mistake in signature verification in audit part.
